mteb/cache.py

```
@@ -275,6 +279,160 @@ def download_from_remote(
+            "http://github.com/", ""
+        )
+
+        url = f"https://raw.githubusercontent.com/{repo_path}/{branch}/{filename}"
+        logger.info(f"Downloading cached results from {url}")
+
+        # Step 1: Download with validation
+        max_size_bytes = max_size_mb * 1024 * 1024
+
+        try:
+            response = requests.get(url, timeout=timeout)
+            response.raise_for_status()
+
+            # Validate content-type header
+            content_type = response.headers.get("content-type", "").lower()
+            if content_type and not any(
+                ct in content_type
+                for ct in [
+                    "application/gzip",
+                    "application/octet-stream",
+                    "application/x-gzip",
+                ]
+            ):
+                logger.warning(
+                    f"Unexpected content-type: {content_type}, continuing anyway..."
+                )
+
+            # Validate file size
+            content_length = len(response.content)
+            if content_length > max_size_bytes:
+                raise ValueError(
+                    f"Downloaded file too large: {content_length} bytes (max: {max_size_bytes})"
+                )
+
+            logger.info(
+                f"HTTP request successful, content length: {content_length} bytes"
+            )
+            content = response.content
+
+        except requests.exceptions.Timeout as e:
+            logger.error(f"HTTP timeout after {timeout}s: {e}")
+            raise
+        except requests.exceptions.ConnectionError as e:
+            logger.error(f"HTTP connection error: {e}")
+            raise
+        except requests.exceptions.HTTPError as e:
+            logger.error(f"HTTP error {response.status_code}: {e}")
+            raise
+        except Exception as e:
+            logger.error(f"Unexpected HTTP error: {type(e).__name__}: {e}")
+            raise
```

if we just raise the error directly after I think we gain a lot from catching the error

---

Yea same as above - I think it is cleaner to not catch the error and just let it be raised